src/AutoRest.CSharp/Output/Models/Client.cs

```diff
@@ -119,5 +121,54 @@ private IEnumerable<ClientMethod> BuildMethods()
                 }
             }
         }
+
+        private IEnumerable<Parameter> GetRequiredParameters()
+        {
+            List<Parameter> parameters = new List<Parameter>();
+
+            if (RestClient.Parameters.TryGetRequiredParameters(out var requiredParameters))
```

Comment:
Can we use the `p.DefaultValue` check only?

Reply:
Yeah, it would work.

Reply:
So do we still need the TryGetRequiredParameters call?

Reply:
Ahh, updated `GetOptionalParameters()` method, forgot to update this one.